fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushDownUnnestInProject.java

```diff
@@ -0,0 +1,138 @@
+                                + "unnest function has error, function count is %d, pushed down count is %d",
+                        toBePushedDown.size(), fields.size()));
+                for (int i = 0; i < fields.size(); ++i) {
+                    Slot remainExpr = slotTripletMap.get(toBePushedDown.get(i)).remainExpr;
+                    newProjects.add(new Alias(remainExpr.getExprId(), new StructElement(
+                            outputSlot, new StringLiteral(fields.get(i).getName())), remainExpr.getName()));
+                }
+            } else {
+                Slot remainExpr = slotTripletMap.get(toBePushedDown.get(0)).remainExpr;
+                newProjects.add(new Alias(remainExpr.getExprId(), outputSlot, remainExpr.getName()));
+            }
+            Map<Slot, Expression> replaceMap = ExpressionUtils.generateReplaceMap(newProjects);
+            List<NamedExpression> newOutputs = context.normalizeToUseSlotRef(outputs);
+            return project.withProjectsAndChild(ExpressionUtils.replaceNamedExpressions(newOutputs, replaceMap),
+                    new LogicalGenerate(ImmutableList.of(newFunction), ImmutableList.of(outputSlot), project.child()));
+        }).toRule(RuleType.PUSH_DOWN_UNNEST_IN_PROJECT);
+    }
+
+    private boolean containUnnestFunction(List<NamedExpression> expressions) {
+        for (NamedExpression expr : expressions) {
+            if (expr.containsType(Unnest.class)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private Unnest validateAndMergeUnnest(List<Unnest> functions) {
+        // TODO: PG only support ARRAY type, and explode_map and explode_bitmap only support 1 argument in doris
+        // so we only allow array for now, too
+        int typeCounter = 0;
+        int size = functions.size();
+        List<Expression> expressions = new ArrayList<>(size);
+        for (Unnest unnest : functions) {
+            if (unnest.children().size() > 1) {
+                throw new AnalysisException("UNNEST function can have more than one arguments only in from clause");
+            }
+            Expression expr = unnest.child(0);
+            DataType dataType = expr.getDataType();
+            if (dataType.isArrayType()) {
+                typeCounter += 1;
+            } else if (dataType.isMapType()) {
+                typeCounter += 2;
+            } else if (dataType.isBitmapType()) {
+                typeCounter += 3;
+            }
+            expressions.add(expr);
+        }
+        // TODO: remove this after doris support multiple arguments for explode_map and explode_bitmap
+        if (expressions.size() > 1 && typeCounter != 1 * size) {
+            throw new AnalysisException("multiple UNNEST functions in same place must have ARRAY argument type");
+        }
+        if (typeCounter == 1 * size || typeCounter == 2 * size || typeCounter == 3 * size) {
```

Comment:
what's this if mean? add comment. if u want check all argument is same type, this impl is not very intuitive